UG-834 Create list variant tracking spec

meel-io opened the pull request:
### Description

The save article overlay variant will use the same tracking events as the current experience apart from an event to track the new createList action. This change adds a function to trigger the `create-list-success` event and calls it after the list is created.

UPDATE: This PR is also adds new custom events for create list, add article to list and remove article from list actions on the save article variant. These events include the additional properties `amplitudeExploratory` and `teamName` to have them appear on Amplitude's exploratory space, and we'll use them to build a dashboard for this feature on Amplitude

### How to test this code?

- Check out this branch locally and run: 
  - `npm link`
  - `bower link`
- Check out `next-article` and run:
  - `npm link @financial-times/n-myft-ui`
  - `bower link n-myft-ui`
- Build and run `next-article`
  - `make build run`
- Go to FT Toggler and select the variant on the [professorLists flag](https://toggler.ft.com/#professorLists)
- Go to [myFT section](https://www.ft.com/myft/lists) and make sure you don't have any lists created. 
- Go to [an article](https://local.ft.com:5050/content/aa9caaad-c19f-4451-b13b-e514b9c284a5) and click on the `Save`  button on the Share Navigation bar
- Open the network tab and confirm that these events are triggered through their respective actions
  - Create a new list: `myFT:create-list-success`
  - Remove article from list: `myFT:unsave`
  - Add article to list: `myFT:add-to-list-success`
- Confirm that these events are triggered through their respective actions and they include the fields `amplitudeExploratory` and `teamName` on the payload's `context` property
  - Create a new list: `professorList:create-list`
  - Remove article from list: `professorList:remove-from-list`
  - Add article to list: `professorList:add-to-list`

leannecornish-ft commented on myft/ui/save-article-to-list-variant.js:
```diff
@@ -23,6 +23,7 @@ export default async function openSaveArticleToListVariant (name, contentId) {
 					overlayContent.insertAdjacentElement('afterbegin', listElement);
 					const announceListContainer = document.querySelector('.myft-ui-create-list-variant-announcement');
 					announceListContainer.textContent = `${list} created`;
+					triggerCreateListEvent(contentId)
```
```suggestion
					triggerCreateListEvent(contentId);
```
Nitpick: because eslint was complaining 😁

leannecornish-ft reviewed (approved):
It looks good to me 👍 

Praise: great instructions on how to test, thank you 🙌 

Aside: unrelated to your change but while testing, I noticed that the article does not get removed from the list when it is unsaved - not sure if you're already aware?

meel-io commented:
> It looks good to me 👍
> 
> Praise: great instructions on how to test, thank you 🙌
> 
> Aside: unrelated to your change but while testing, I noticed that the article does not get removed from the list when it is unsaved - not sure if you're already aware?

Thanks for the review. I fixed the eslint issue, good spot 👁️ 
I added an extra set of events that will be needed to create a dashboard on Amplitude and an extra step on the `How to test this code?` section to confirm it's working

fenglish reviewed (approved):
Looks good to me
